Remotion/Web/ClientScript/Scripts/DatePicker.ts

```diff
@@ -150,67 +159,80 @@ class DatePicker
       if (DatePicker._datePicker_repositionTimer)
         clearTimeout (DatePicker._datePicker_repositionTimer);
 
-      if (DatePicker._datePicker_current && DatePicker._datePicker_current.DatePicker == datePicker && datePicker.is (':visible'))
+      if (DatePicker._datePicker_current && DatePicker._datePicker_current.DatePicker == datePicker && LayoutUtility.IsVisible(datePicker))
       {
-        DatePicker.ApplyPosition (datePicker, $(button));
+        DatePicker.ApplyPosition (datePicker, button);
         DatePicker._datePicker_repositionTimer = setTimeout (repositionHandler, DatePicker._datePicker_repositionInterval);
       }
     };
 
-    DatePicker.ApplyPosition (datePicker, $(button));
+    DatePicker.ApplyPosition (datePicker, button);
     DatePicker._datePicker_repositionTimer = setTimeout(repositionHandler, DatePicker._datePicker_repositionInterval);
 
     return datePicker;
   }
 
-  private static ApplyPosition (datePicker: JQuery, button: JQuery): void
+  private static ApplyPosition (datePicker: HTMLElement, button: HTMLElement): void
   {
     var datePickerLeft;
     var datePickerTop;
     var datePickerWidth;
     var datePickerHeight;
 
-    var body = $ ('body');
-    var left = $(button).offset().left;
-    var top = $(button).offset().top;
+    var body = document.body;
+    var left = LayoutUtility.GetOffset(button).left;
+    var top = LayoutUtility.GetOffset(button).top;
 
     //  Adjust position so the date picker is shown below 
     //  and aligned with the right border of the button.
-    datePicker.css('left', Math.max (0, left - datePicker.width() + $(button).width()));
-    datePicker.css('top', Math.max (0, top + $(button).height()));
-    datePickerLeft = datePicker.offset().left;
-    datePickerTop = datePicker.offset().top;
-    datePickerWidth = datePicker.width();
-    datePickerHeight = datePicker.height();
+    datePicker.style.left = Math.max (0, left - LayoutUtility.GetWidth(datePicker) + LayoutUtility.GetWidth(button)) + 'px';
+    datePicker.style.top = Math.max (0, top + LayoutUtility.GetHeight(button)) + 'px';
+    datePickerLeft = LayoutUtility.GetOffset(datePicker).left;
+    datePickerTop = LayoutUtility.GetOffset(datePicker).top;
+    datePickerWidth = LayoutUtility.GetWidth(datePicker);
+    datePickerHeight = LayoutUtility.GetHeight(datePicker);
 
     //  Re-adjust the button, in case available screen space is insufficient
-    var visibleBodyTop = body.scrollTop();
-    var visibleBodyHeight = $(window).height();
+    var visibleBodyTop = body.scrollTop;
```

Comment:
In DropDownMenu, the transformation was 
`$(document).scrollTop()` to `window.pageYOffset`

I take it, scroll on window and scroll on document use different APIs? And should we replace window-scroll with document-scroll in the other files?

Reply:
@a-ctor here's an unanswered review-question left open.

Reply:
That depends on what the scroll container is and, naturally, on what you are trying to measure. If whole page is never bigger than the screen (as is the case on our test pages) the windows pageOffset values will always be 0. So the action we should take depends on the css of the page.

Fun fact: On the individual controls test page both window and the body element have no scroll and will always have (0,0) as scroll value.